server/commands/check.go

```diff
@@ -75,13 +76,60 @@ func (query *CheckQuery) Execute(ctx context.Context, req *openfgapb.CheckReques
 
 	typesys := typesystem.New(model)
 
-	rc := newResolutionContext(req.GetStoreId(), model, tk, contextualTuples, resolutionTracer, utils.NewResolutionMetadata(), &circuitBreaker{breakerState: false})
+	if err := validation.ValidateObject(typesys, tk); err != nil {
+		return nil, serverErrors.HandleTupleValidateError(err)
+	}
 
-	err = validation.ValidateTuple(typesys, tk)
-	if err != nil {
+	if err := validation.ValidateRelation(typesys, tk); err != nil {
+		return nil, serverErrors.HandleTupleValidateError(err)
+	}
+
+	if err := validation.ValidateUser(typesys, tk); err != nil {
 		return nil, serverErrors.HandleTupleValidateError(err)
 	}
 
+	if typesys.GetSchemaVersion() == typesystem.SchemaVersion1_1 {
+		g := graph.BuildConnectedObjectGraph(typesys)
```

Comment:
Actually, now that I'm thinking about this more.. I think this block of code is completely unnecessary. The graph evaluation will still skip invalid tuples, but we just want to make sure we don't return an error on indirect types.

So I should be able to omit this block of code and just have the object, relation, and user validation checks above and we should be G2G. I will remove this and verify that's alright.